libmozevent/pulse.py

```diff
@@ -92,30 +93,33 @@ class PulseListener(object):
 
     def __init__(
         self,
-        output_queue_name,
-        queues_routes: List[PulseBinding],
+        queues_routes: Dict[BusQueue, List[PulseBinding]],
         user,
         password,
         virtualhost="/",
     ):
-        self.queue_name = output_queue_name
         self.queues_routes = queues_routes
         self.user = user
         self.password = password
         self.virtualhost = virtualhost
 
     def register(self, bus):
         self.bus = bus
-        self.bus.add_queue(self.queue_name)
+        for name in self.queues_routes:
+            self.bus.add_queue(name)
 
     async def connect(self):
-        protocol = await create_pulse_listener(
-            self.user,
-            self.password,
-            self.queues_routes,
-            self.got_message,
-            self.virtualhost,
-        )
+        listeners = [
```

Comment:
I think it'd be better to have a single pulse connection, to reduce resource usage.

Reply:
So a single callback that find the good route from the message routing information ?
It would be more stable pulse-wise, but may lead to bugs if we have a message that matches multiple routes

Reply:
https://www.cloudamqp.com/blog/2018-01-19-part4-rabbitmq-13-common-errors.html suggests having one connection per process and one channel per thread.
Perhaps we could have one connection only, and one channel for each listener.

Reply:
Or you can use the exchange_name info from the envelope.

Or yet another option could be to use a consumer_tag in basic_consume (which will also be available in the envelope), so each consumer can be tagged.